mono/mini/mini.c

```diff
@@ -3685,6 +3685,8 @@ mini_method_compile (MonoMethod *method, guint32 opts, MonoDomain *domain, JitFl
 		mono_local_deadce (cfg);
 	if (cfg->opt & MONO_OPT_ALIAS_ANALYSIS)
 		mono_local_alias_analysis (cfg);
+	if (cfg->opt & MONO_OPT_DIVREM && cfg->has_divisions)
+		mono_local_div_imm_removal (cfg);
```

Comment:
@vargaz @kumpera 

The main code is finished but there are still some problems with the structure.

First of all, as bernhard pointed out, is there a better place for this ? A general purpose optimization pass ? I don't think cprop is a good idea. I'm not sure whether the peephole pass is made for such general changes (like creating new basic blocks and vregs)

Second, the pass depends on cprop to propagate immediates and reduce divs to div_imms. This means it also has to run after the long_decomposition. Normally we should emit the decomposed code, but for development, it was easier this way. This would make the code tougher to read, might require some weird defs etc. Approve of emitting directly the decomposed form ?

Third, on platforms where we emulate division (arm), the division is emulated instantly, at IR emitting time (mono_decompose_opcode) so it doesn't give us any chance to optimize the division. Any reasons for this to happen and suggestions where to move the emulation ? Separate pass, existing pass ? Normally we would want to run it as late as possible before emitting native code.